libs/service-portal/settings/access-control/src/screens/GrantAccess/GrantAccess.tsx

```diff
@@ -58,23 +58,28 @@ function GrantAccess() {
   ) => {
     const value = e.target.value
     if (value.length === 10 && kennitala.isValid(value)) {
-      getDelegation({ variables: { input: { toNationalId: value } } })
+      getIdentity({ variables: { input: { nationalId: value } } })
+      if (identity?.nationalId === value) {
+        setValue('name', identity.name)
+      }
+    } else {
+      setValue('name', '')
     }
   }
 
   useEffect(() => {
-    if (authDelegation) {
-      setValue('name', authDelegation.toName)
+    if (identity) {
```

Comment:
Can we check `toNationalId` here to check if it's still the same as `identity.nationalId`?

Reply:
Would be an unnecessary check.

Reply:
What if I type in the tenth digit, and then figure out I did a mistake and remove some digits before the graphql request resolves?

Reply:
Wow, you're right. Good spot!

Reply:
Impl: https://github.com/island-is/island.is/pull/4447/commits/3652c269da95a8256b8b8755c307c05fb043ea9c